### DISCUSSION TITLE
Pagination

### DISCUSSION THREAD

**Discussion by cindyebi** [visual-framework/vf-core#1568] [Proposals]
We have the basic version of the pagination component:
https://latest.visual-framework.dev//components/vf-pagination/index.html?q=pagination

<img width="1121" alt="Screenshot 2021-06-03 at 11 16 57" src="https://user-images.githubusercontent.com/75834903/120628975-37141e00-c45d-11eb-95e3-f5276885bd25.png">

How could we improve this component for different use cases?
Ideas are to include or create new variants for:
• View per page
• Jump to a specific page
• Small or compact variant

@nitinja showed a good pagination reference yesterday: https://ant.design/components/pagination/ 

There're also some minor visual changes required on the current (basic) version: hover and active link colours are grey instead of interactive blue. 

NNGroup's Pagination guidelines: https://www.nngroup.com/articles/item-list-view-all/

**Comment by khawkins98** [comment]
My concern with Ant's approach is that there are so many options: that's difficult to support technically and from a documentation approach. 

I think we'd be able to deliver more value in having a smaller and more consistent variant set.

I also like IBM's Carbon pager as it resolves many RWD questions by compacting the `page 1, 2, 3, 4, 5, 6, 7` readout into a select element. 

We'll also should try to leverage some `vf-form` code for the select elements, and perhaps others.

**Comment by nikiforosk** [comment]
A couple of thoughts following a discussion with @prakashebi about [the contrast issues](https://trello.com/c/ULllvb9o) with the way [vf-pagination](https://stable.visual-framework.dev/components/vf-pagination/) is implemented in the Training site, e.g. in the [on-demand training listing](https://www.ebi.ac.uk/training/on-demand?facets=&page=1&query=):

![vf-pagination-ondemand-p1-Screenshot 2022-08-09 at 14 46 02](https://user-images.githubusercontent.com/20280139/183665416-445c1cc2-e4b5-4664-bcf4-bd8e3cc5319e.png)

Links such as "2" above use `#707372` (i.e. `--vf-color--neutral--600`) which [passes the contrast check at the AA level](https://webaim.org/resources/contrastchecker/?fcolor=707372&bcolor=FFFFFF) but not at the AAA level. 

The problem with that is that we cannot use a lighter gray colour for "disabled" links such as "Previous" in the above example. This is because `--vf-color--neutral--500` (which corresponds to `#8d8f8e`) [completely fails the contrast checks for normal text at the AA level](https://webaim.org/resources/contrastchecker/?fcolor=8D8F8E&bcolor=FFFFFF). Given that the text is 16px it would classify as normal text.

I couldn't figure out in the description of this component which colour is meant to be used for "disabled" links. On the training site we're using `#808080` which isn't a vf-color and also [fails the contrast check for normal text](https://webaim.org/resources/contrastchecker/?fcolor=808080&bcolor=FFFFFF).

I think that in this context we could use the link colour `#3B6FB6` for links and the Secondary Text Colour `#373A36` for the disabled states. This is a quick mockup from changing the colours on the browser:

![vf-pagination-v01-Screenshot 2022-08-09 at 11 40 11](https://user-images.githubusercontent.com/20280139/183668627-a5a70ccc-5977-452b-a44f-f3abd3ca546e.png)

I feel that this better contrasts links with non links (i.e. text) using conventions that we already have:

![vf-pagination-v01-Screenshot 2022-08-09 at 11 37 54](https://user-images.githubusercontent.com/20280139/183668598-d6f25875-0a14-454a-aa3f-195b615ae1c6.png)

Further to this, I think we could also get rid of the current mouseover effect, e.g. on page 3 (which is the same as the active state of page 1):
![vf-pagination-mouseover-3-Screenshot 2022-08-09 at 15 05 23](https://user-images.githubusercontent.com/20280139/183670187-58bc11a9-c017-4da7-8aa9-72077f7636e5.png)

We can use the underline of normal vf-links for mouseover and signify the active state simply by using **bold** Primary Text Colour
`#1a1c1a`. The pagination of the google search results is very similar to this:
![google-search-pagination-Screenshot 2022-08-09 at 15 08 33](https://user-images.githubusercontent.com/20280139/183671026-80824181-a250-47af-881d-f726ac28064a.png)
